integrations/access/servicenow/config.go

```diff
@@ -69,6 +70,13 @@ func (c *Config) CheckAndSetDefaults() error {
 	return nil
 }
 
+func (c *Config) GetTeleportClient(ctx context.Context) (teleport.Client, error) {
+	if c.Client != nil {
+		return c.Client, nil
+	}
+	return c.BaseConfig.GetTeleportClient(ctx)
```

Comment:
I don't think it's racy because we never write c.Client back. There is only one case that currently sets c.Client: if your plugin is started as a hosted plugin, this is set on configuration creation and nothing should write it later. 

If you call c.GetTeleportClient() multiple times sequentially or concurrently you will either always get c.Client immediately or get multiple new clients. This is also the case for other plugins.

I completely agree that we should centralize client management behind a sync.OnceValue(), but I'd prefer to do this in a proper refactoring (because we need to refactor the mess that is plugin configuration; see: [the slack thread](https://gravitational.slack.com/archives/C01UV6NKQTS/p1710786103891789?thread_ts=1710441145.870529&cid=C01UV6NKQTS) that triggered this PR).